osu.Game/OsuGame.cs

```diff
@@ -313,6 +309,36 @@ private void load()
         /// <param name="beatmapId">The beatmap to show.</param>
         public void ShowBeatmap(int beatmapId) => waitForReady(() => beatmapSetOverlay, _ => beatmapSetOverlay.FetchAndShowBeatmap(beatmapId));
 
+        /// <summary>
+        /// Show a beatmap set as an overlay.
+        /// </summary>
+        /// <param name="setId">The set to display.</param>
+        public void ShowBeatmapSet(int setId) => waitForReady(() => beatmapSetOverlay, _ => beatmapSetOverlay.FetchAndShowBeatmapSet(setId));
+
+        public void SelectBeatmap(int beatmapId) => PresentBeatmap(GetBeatmapFromId(beatmapId).BeatmapSet, b => b.OnlineBeatmapID == beatmapId);
+
+        public void SelectBeatmapSet(int setId) => PresentBeatmap(GetBeatmapSetFromId(setId).BeatmapSet);
+
+        public BeatmapInfo GetBeatmapFromId(int beatmapId) => BeatmapManager.QueryBeatmap(b => b.OnlineBeatmapID == beatmapId);
+
+        public BeatmapInfo GetBeatmapSetFromId(int setId) => BeatmapManager.QueryBeatmap(b => b.BeatmapSet.OnlineBeatmapSetID == setId);
+
+        public int GetBeatmapIdFromLink(LinkDetails link)
+        {
+            if (link.Argument != null && int.TryParse(link.Argument.Contains('?') ? link.Argument.Split('?')[0] : link.Argument, out int beatmapId))
+                return beatmapId;
+
+            return 0;
+        }
+
+        public int GetBeatmapSetIdFromLink(LinkDetails link)
+        {
+            if (int.TryParse(link.Argument, out int setId))
+                return setId;
+
+            return 0;
+        }
```

Comment:
these also have no reason to be in `OsuGame`. it should be the responsibility of a link to provide this data, and I'd say we're nearing the point in which we might want to have an actual type hierarchy of `Link`s so that we can properly read these as plain old properties instead of parsing strings.

if that is the direction we are to take things then that is a separate preparatory refactoring PR.